web/packages/teleport/src/services/user/user.test.ts

```diff
@@ -370,3 +371,79 @@ test('makeTraits', async () => {
     color: [],
   });
 });
+
+test('excludeUserFields when updating user', async () => {
+  // we are not testing the reply, so reply doesn't matter.
+  jest.spyOn(api, 'put').mockResolvedValue({} as any);
+
+  const userReq: User = {
+    name: 'name',
+    roles: [],
+    traits: blankTraits,
+    allTraits: {},
+  };
+
+  await user.updateUser(userReq, 'allTraits');
+  expect(api.put).toHaveBeenCalledWith(cfg.api.usersPath, {
+    name: 'name',
+    roles: [],
+    traits: blankTraits,
+  });
+
+  jest.clearAllMocks();
+
+  await user.updateUser(userReq, 'traits');
```

Comment:
minor readability nit: Maybe if we can use full typed name `ExludeUserField.Traits`, then it would be clear while reading the code that the "traits" is supposed to be excluded, and also be away with comments such as `(..., 'traits' /* exclude field */ ...)`.

Reply:
that's great suggestion, thanks!